src/main/java/fi/otavanopisto/kuntaapi/server/controllers/IncidentController.java

```diff
@@ -0,0 +1,131 @@
+package fi.otavanopisto.kuntaapi.server.controllers;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import fi.otavanopisto.kuntaapi.server.id.IncidentId;
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.integrations.IncidentProvider;
+import fi.otavanopisto.kuntaapi.server.integrations.IncidentSortBy;
+import fi.otavanopisto.kuntaapi.server.integrations.SortDir;
+import fi.otavanopisto.kuntaapi.server.utils.ListUtils;
+import fi.metatavu.kuntaapi.server.rest.model.Incident;
+
+@ApplicationScoped
+@SuppressWarnings ("squid:S3306")
+public class IncidentController {
+  
+  @Inject
+  private EntityController entityController;
+  
+  @Inject
+  private Instance<IncidentProvider> incidentProviders;
+  
+  public List<Incident> listIncidents(OrganizationId organizationId, String slug, OffsetDateTime startBefore, OffsetDateTime endAfter, IncidentSortBy sortBy, SortDir sortDir, Integer firstResult, Integer maxResults) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)